Automattic/pocket-casts-ios
# Enable test coverage for pocketcasts target

## Changes

1.  Enable xcode test coverage reports
2.  Enable xcode side bar test coverage

| 1 | 2 |
|:-------:|:-------:|
| ![image1](https://github.com/Automattic/pocket-casts-ios/assets/18557720/32462279-e803-41be-b0b2-fc1c6c02b962) | ![image2](https://github.com/Automattic/pocket-casts-ios/assets/18557720/7da7248f-dbfa-45bc-9537-4ff245ab2ce8) |

## Why

Apple's test coverage reports allow us to see at a glance what code has been tested and where we're lacking test coverage. The overall figure is also a valuable metric to track whether sufficient testing is being done for a given project, component or class which helps inform team policy decisions regarding testing or testability enablers such as modularization, or dependency injection.

The side bar test coverage can help developers when writing tests to see what parts of a class they're not testing and give us ideas for additional tests.

While it's a valuable metric/developer tool it shouldn't be used as a golden bullet. The absence of coverage is useful in itself but just because code is covered doesn't mean it's well tested as we need to test the code's behavior in a variety of circumstances and this involves writing much more than a single test per function. I'm also of the opinion that 100% coverage is not something any healthy project should aim for.

## To test

* Run all unit tests
* No new warnings or errors should appear
* Go to the report navigator (cmd 9) and click coverage. There should be a full test coverage report showing 1.5% total coverage.
* Check that 3rd party code has not been included within the coverage figures
* Check that all 1st party code has been included within the coverage figures


## Checklist
